Removed PillowTestCase helper class

radarhere opened the pull request:
Helps #4193

Converts all uses of PillowTestCase to pytest, so that PillowTestCase is no longer needed.

hugovk commented:
![image](https://user-images.githubusercontent.com/1324225/77827242-3414f380-711d-11ea-82a7-892e8cd3fc2b.png)

This Codecov "+-0.01%" is newish (I think it was only a plus or a minus earlier), and a +- hundredth of a percent is pretty meaningless.

Shall we add a `threshold` to stop it?

> Allow the coverage to drop by `X%`, and posting a `success` status.

https://docs.codecov.io/docs/commit-status#section-threshold

Perhaps `0.01%` or `0.1%`? And if so, here or in another PR?

radarhere commented:
I've added the conversion of unittest mock to pytest monkeypatch, after receiving the suggestion from #4488

Regarding the threshold, yes, I guess we can. I think the diff hit coverage might be more meaningful statistic of the two. I've added that commit as well.
